### PR TITLE
feat: Add BTC related metrics

### PR DESCRIPTION
**Pull request by gitferry**
This PR adds the following Prometheus metrics:

Vigilante submitter: For every submitted transaction export the following. These metrics should be emitted every time a new checkpoint is submitted.
* Tx hashes of checkpoint segments (2)
* Fees for each segment
* Epoch number
* Timestamp
* total available balance sum
* re-send checkpoint interval (emitted when the submitter is first created)
* Number of invalid checkpoints (either epoch or status is invalid) for alerting
* Number of resent checkpoints (triggered if the resend_interval is passed)
* Number of failed resent checkpoints for alerting

Vigilante reporter: For every BTC block export the following. These metrics should be emitted every time a new BTC block / checkpoint is sent to Babylon.
* When a BTC block is submitted
    * BTC block timestamp
* When a BTC checkpoint is submitted
    * Tx hashes of the checkpoint segments
    * The Babylon epoch that it corresponds to
    * BTC block height

### CODE REVIEW

File: metrics/submitter.go [hunk 1]
```diff
@@ -14,6 +14,60 @@ type SubmitterMetrics struct {
...
+	FailedResentCheckpointsCounter        prometheus.Counter
+	NewSubmittedCheckpointSegmentGaugeVec *prometheus.GaugeVec
+}
+
+func NewRelayerMetrics(registry *prometheus.Registry) *RelayerMetrics {
+	registerer := promauto.With(registry)
+
+	metrics := &RelayerMetrics{
+		ResendIntervalSecondsGauge: registerer.NewGauge(prometheus.GaugeOpts{
+			Name: "vigilante_submitter_resend_intervals",
```
Comment (vitsalis) [inline metrics/submitter.go:31]:
```suggestion
			Name: "vigilante_submitter_resend_interval",
```

File: submitter/relayer/relayer.go [hunk 2]
```diff
@@ -102,9 +108,19 @@ func (rl *Relayer) SendCheckpointToBTC(ckpt *ckpttypes.RawCheckpointWithMeta) er
...
+			rl.metrics.FailedResentCheckpointsCounter.Inc()
 			return fmt.Errorf("failed to re-send the second tx of the checkpoint %v: %w", rl.lastSubmittedCheckpoint.Epoch, err)
 		}
 
+		// record the metrics of the resent tx2
+		rl.metrics.NewSubmittedCheckpointSegmentGaugeVec.WithLabelValues(
+			strconv.Itoa(int(ckptEpoch)),
+			"1",
+			resubmittedTx2.TxId.String(),
+			fmt.Sprintf("%d Satoshis", resubmittedTx2.Fee),
```
Comment (vitsalis) [inline submitter/relayer/relayer.go:120]: No need to do formatting here, the consumer can do formatting themselves the way they want.

File: submitter/relayer/relayer.go [hunk 3]
```diff
@@ -253,6 +269,20 @@ func (rl *Relayer) convertCkptToTwoTxAndSubmit(ckpt *ckpttypes.RawCheckpointWith
...
+		strconv.Itoa(int(ckpt.Ckpt.EpochNum)),
+		"0",
+		tx1.Tx.TxHash().String(),
+		fmt.Sprintf("%d Satoshis", tx1.Fee),
+	).SetToCurrentTime()
+	rl.metrics.NewSubmittedCheckpointSegmentGaugeVec.WithLabelValues(
+		strconv.Itoa(int(ckpt.Ckpt.EpochNum)),
+		"1",
+		tx2.Tx.TxHash().String(),
+		fmt.Sprintf("%d Satoshis", tx2.Fee),
```
Comment (vitsalis) [inline submitter/relayer/relayer.go:283]: Same for the above.

**Review by vitsalis — APPROVED** [review]
Looks good overall. @filippos47 does this cover your needs?

**Review by filippos47 — APPROVED** [review]
Looks good to me!

The only metric missing compared to the original spec is `List of available UTXOs`, which would be emitted by the submitter. I can obtain this information from the bitcoind exporter that we're using, so it's not critical to add to vigilante at this point.

**Comment by gitferry** [comment]
Thanks, @filippos47 for the reminder. Yeah, emitting the `List of available UTXOs` is a bit tricky for me. Glad to know that we can obtain the data from the bitcoins exporter.

**Pull request merged**